core/src/Instance/B2CAuthority.cs

```diff
@@ -44,8 +45,9 @@ internal B2CAuthority(string authority, bool validateAuthority) : base(authority
                 throw new ArgumentException(CoreErrorMessages.B2cAuthorityUriInvalidPath);
             }
 
-            CanonicalAuthority = string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/{2}/{3}/", authorityUri.Authority,
+            CanonicalAuthority = string.Format(CultureInfo.InvariantCulture, B2CCanonicalAuthorityTemplate, authorityUri.Authority,
                 pathSegments[0], pathSegments[1], pathSegments[2]);
```

Comment:
this behaves differently from UpdateTenentId, is that on purpose? Could you simply call UpdateTenentId here and get the expected CanonicalAuthority out?

Reply:
this I do not get. then why is it implemented in this class?

---
In reply to: [216100716](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1246#discussion_r216100716) [](ancestors = 216100716)

Reply:
that part should be refactored ))

Reply:
I can do that if we have time, do we ?